### PR TITLE
update haproxy exporter default to 0.9.0, fix options and allow unix …

### PR DESCRIPTION
**Pull request by dynek**
#### Pull Request (PR) description
- Update default haproxy exporter version from 0.7.1 (released on 2016-10-12) with 0.9.0 (released on 2018-01-23)
- Fix options since they moved to kingping: https://github.com/prometheus/haproxy_exporter/commit/cb544ab842c398242fc5e22053da49dda3058aa0
- Allow polling from Unix socket
- Remove Stdlib::HTTPSUrl from Variants as Stdlib::HTTPUrl covers both http and httpS

### CODE REVIEW

**Comment by dynek** [comment]
Failed because of:
```
debian8-64-1 10:45:16$ cat /etc/resolv.conf
  Attempting ssh connection to 0.0.0.0, user: root, opts: {:password=>"root", :port=>"53", :forward_agent=>false}
  Warning: Try 1 -- Host 0.0.0.0 unreachable: Errno::ECONNRESET - Connection reset by peer
  Warning: Trying again in 3 seconds
```

File: manifests/haproxy_exporter.pp [hunk 1]
```diff
@@ -73,15 +73,15 @@
 #  The binary release version
 
 class prometheus::haproxy_exporter(
-  Variant[Stdlib::HTTPUrl, Stdlib::HTTPSUrl] $cnf_scrape_uri,
+  Variant[Stdlib::HTTPUrl, Pattern[/unix:(?:\/.+)+/]] $cnf_scrape_uri,
```
Comment (bastelfreak) [inline manifests/haproxy_exporter.pp:76]: * Why did you remove the https type?
* Would https://github.com/puppetlabs/puppetlabs-stdlib/blob/master/types/unixpath.pp work as well for a unix socket?

File: data/defaults.yaml [hunk 2]
```diff
@@ -110,7 +110,7 @@ prometheus::haproxy_exporter::group: 'haproxy-exporter'
 prometheus::haproxy_exporter::package_ensure: 'latest'
 prometheus::haproxy_exporter::package_name: 'haproxy_exporter'
 prometheus::haproxy_exporter::user: 'haproxy-user'
-prometheus::haproxy_exporter::version: '0.7.1'
+prometheus::haproxy_exporter::version: '0.9.0'
```
Comment (bastelfreak) [inline data/defaults.yaml:113]: can you please add an acceptance test that first explictly installs 0.7.1 and afterwards upgrades it to 0.9.0? We already have a few tests in this repo that do the same for other exporters.

**Comment by bastelfreak** [comment]
Hi @dynek, thanks for the PR. Can you take a look at the used email address in your commits? It isn't associated with your github account.

**Comment by dynek** [comment]
> Hi @dynek, thanks for the PR. Can you take a look at the used email address in your commits? It isn't associated with your github account.

My bad - Is it OK for you now?

File: manifests/haproxy_exporter.pp (hunk 1 above)
Comment (dynek) [reply manifests/haproxy_exporter.pp:76]: I removed https type because Stdlib::HTTPUrl covers both http and httpS, see https://github.com/puppetlabs/puppetlabs-stdlib#user-content-stdlibhttpurl

Comment (dynek) [reply manifests/haproxy_exporter.pp:76]: Unixpath doesn't work for path as it doesn't allow ":" in the pattern.
Example of what I'm currently using: `unix:/var/lib/haproxy/stats`

File: data/defaults.yaml (hunk 2 above)
Comment (dynek) [reply data/defaults.yaml:113]: Makes sense. I'll check this asap.

**Comment by dynek** [comment]
@bastelfreak finally added acceptance tests but all CI tests are failing because `The program 'bundle' is currently not installed`

**Comment by Dan33l** [comment]
> @bastelfreak finally added acceptance tests but all CI tests are failing because `The program 'bundle' is currently not installed`

Yes it looks a travis issue. Several modules are concerned by this issue.

**Comment by dynek** [comment]
There we go, checks have passed.

**Comment by dynek** [comment]
@bastelfreak anything else I can do?

**Comment by bastelfreak** [comment]
Thanks for the PR @dynek !
